src/compile.rs

```diff
@@ -1695,52 +1688,87 @@ fn interpret_associate(
-                                annotations: Annotations {
-                                    annotations: vec![Annotation {
-                                        name: "associate".into(),
-                                        arguments: vec![Argument::List(vec![class_name])],
-                                    }],
-                                },
-                            },
-                        ))));
-                    }
-                    Err(e) => errors.add_error(e),
+        // Okay, so this is tricky, because we might have multiple parents.  Eg foo1.bar and
+        // foo2.bar.
+        // I *think* we need to find *all* of them.  They probably should *all* exist (which means
+        // not rewriting to children on inheritance?).  Then if they don't *all* exist, we error.
+        // Else, we create with multiple parents.
+        // I think the problem with the above logic is that we already *know* the parents.  We have
+        // a resource name, and we have some parents.  We know the parents, because they're an
+        // argument to this function.  Rather than finding "res", here, we need to find parent.res
+        // for each parent.
+        // Once we do that, we don't need separate NestedAssociate, right?  Just regular associate
+        // and we're never looking for a global type, unless there *isn't* a parent that we're
+        // inheriting it from
+        // The problem with the above logic is that parent.resource may not exist yet (ie, we're
+        // generating it here, so we've made AST structures, but it's not in the TypeMap).  We
+        // should always have one that already exists, but it might be just `resource` or it might
+        // be `ancestor.resource`, and we don't know which, or which ancestor.  And multiple
+        // inheritance is possible.
+        // Okay, so we use the associated resources as planned.  Basically exactly the original
+        // logic.  The objection to the objection is that the parents aren't necessarily the ones
+        // that have the real resources
+        // What the above is missing is that for dom.res, res may or may not exist.  Actually,
+        // existance is semi-irrelevant.  We need to know if we associated the one that exists,
+        // specifically.  Which, I guess means that real_parent_resources maybe also need to be
+        // parents?  The alternative, which may be cleaner is just to save in AssociatedResource
+        // whether we have the base or not.
+
+        let mut classes = Vec::new();
+        for real_parent_resource in res.get_class_names() {
+            match types.get(&real_parent_resource) {
+                Some(class) => classes.push(class),
+                None => {
+                    // TODO: This is kind of a weird error in the nested case, since the full name
+                    // doesn't exist in source
+                    // I think AssociatedResource needs to be augmented to handle having the
```

Comment:
Candidate for issue?

Reply:
Hmm, the error actually looks fine to me now...  I probably wrote that comment earlier in development and should have deleted it.  The second two lines about augmenting AssociatedResource is definitely done.

Probably what I should do here is add another error test case to exercise this code path and make sure it looks right, then delete the comment.

Reply:
I was going to do what I said above, but I actually don't think this path is reachable in the nested case.  The nested method of association actually creates the resource, so you can't have it not exist.  And even if it were reachable, it should be fine with the AssociatedResource augmentation that is done.  So I'm going to just delete this comment.